hw/bus/drivers/i2c_da1469x/src/i2c_da1469x.c

```diff
@@ -0,0 +1,677 @@
+    .init_node = i2c_da1469x_init_node,
+    .enable = i2c_da1469x_enable,
+    .configure = i2c_da1469x_configure,
+    .read = i2c_da1469x_read,
+    .write = i2c_da1469x_write,
+    .disable = i2c_da1469x_disable,
+};
+
+int
+bus_i2c_da1469x_dev_init_func(struct os_dev *odev, void *arg)
+{
+    struct bus_i2c_dev *dev = (struct bus_i2c_dev *)odev;
+    struct bus_i2c_dev_cfg *cfg = arg;
+    struct i2c_da1469x_dev_data *dd;
+#if MYNEWT_VAL(I2C_DA1469X_STAT)
+    char *stats_name;
+#endif
+    int rc;
+    const struct da1469x_i2c_hw *i2c_hw;
+    static const struct da1469x_dma_config rx_cfg = {
+        .src_inc = false,
+        .dst_inc = true,
+        .priority = 0,
+        .burst_mode = 0,
+        .bus_width = 0,
+    };
+    static const struct da1469x_dma_config tx_cfg = {
+        .src_inc = true,
+        .dst_inc = false,
+        .priority = 0,
+        .burst_mode = 0,
+        .bus_width = 0,
+    };
+
+    BUS_DEBUG_POISON_DEV(dev);
+
+    if (i2c_dev_data[cfg->i2c_num].dev) {
+        return SYS_EALREADY;
+    }
+
+    i2c_dev_data[cfg->i2c_num].dev = dev;
+    i2c_hw = &da1469x_i2c[cfg->i2c_num];
+    dd = &i2c_dev_data[cfg->i2c_num];
+
+    dev->cfg = *cfg;
+
+    da1469x_pd_acquire(MCU_PD_DOMAIN_COM);
+
+    /* Select DIVN clock */
+    CRG_COM->RESET_CLK_COM_REG = (1U << i2c_hw->clk_src_bit);
+    /* Enable clock */
+    CRG_COM->SET_CLK_COM_REG = (1U << i2c_hw->clk_ena_bit);
+
+    /* Abort ongoing transactions and disable I2C */
+    i2c_hw->regs->I2C_ENABLE_REG |= (1 << I2C_I2C_ENABLE_REG_I2C_ABORT_Pos);
+    i2c_hw->regs->I2C_ENABLE_REG &= ~(1 << I2C_I2C_ENABLE_REG_I2C_EN_Pos);
+    while (i2c_hw->regs->I2C_ENABLE_STATUS_REG & I2C_I2C_ENABLE_STATUS_REG_IC_EN_Msk);
+
+    i2c_hw->regs->I2C_CON_REG = (1U << I2C_I2C_CON_REG_I2C_MASTER_MODE_Pos) |
+                                (1U << I2C_I2C_CON_REG_I2C_SPEED_Pos) |
+                                (1U << I2C_I2C_CON_REG_I2C_RESTART_EN_Pos);
+
+    i2c_hw->regs->I2C_INTR_MASK_REG = 0;
+    i2c_da1469x_clear_int(i2c_hw->regs);
+    da1469x_dma_acquire_periph(-1, i2c_hw->dma_periph, dd->dma_chans);
+    da1469x_dma_configure(dd->dma_chans[0], &rx_cfg,
+                          (da1469x_dma_interrupt_cb)i2c_da1469x_dma_rx_isr, dev);
+    dd->dma_chans[0]->DMA_A_START_REG = (uint32_t)&i2c_hw->regs->I2C_DATA_CMD_REG;
+    da1469x_dma_configure(dd->dma_chans[1], &tx_cfg,
+                          (da1469x_dma_interrupt_cb)i2c_da1469x_dma_tx_isr, dev);
+    dd->dma_chans[1]->DMA_B_START_REG = (uint32_t)&i2c_hw->regs->I2C_DATA_CMD_REG;
+
+    mcu_gpio_set_pin_function(cfg->pin_scl, MCU_GPIO_MODE_OUTPUT_OPEN_DRAIN,
+                              i2c_hw->scl_func);
+    mcu_gpio_set_pin_function(cfg->pin_sda, MCU_GPIO_MODE_OUTPUT_OPEN_DRAIN,
+                              i2c_hw->sda_func);
+
+    NVIC_DisableIRQ(i2c_hw->irqn);
+    NVIC_SetVector(i2c_hw->irqn, (uint32_t)i2c_hw->isr);
+    NVIC_SetPriority(i2c_hw->irqn, (1 << __NVIC_PRIO_BITS) - 1);
```

Comment:
I believe the OS startup code already sets the priority to be the lowest possible for all IRQ sources.